src/storage/src/source/postgres/replication.rs

```diff
@@ -209,10 +209,17 @@ pub(crate) fn render<G: Scope<Timestamp = MzOffset>>(
                 .connect_replication(&config.config.connection_context.ssh_tunnel_manager)
                 .await?;
 
+            let metadata_client = connection_config
+                .connect(
+                    "replication metadata",
+                    &config.config.connection_context.ssh_tunnel_manager,
+                )
+                .await?;
+
             tracing::info!(%id, "ensuring replication slot {slot} exists");
             super::ensure_replication_slot(&replication_client, slot).await?;
-            let slot_lsn = super::fetch_slot_resume_lsn(
-                &replication_client,
+            let slot_metadata = super::fetch_slot_metadata(
+                &metadata_client,
```

Comment:
why was it important to switch the client/connection being used here? dont we also want to log the PID of the replication client?

Reply:
It wasn't important to switch the connection, I just wanted to use the nicer type aware API that is only available on the normal client and not the simple query only API that is available in the replication client.